pkg/kubelet/kubelet_pods.go

```diff
@@ -1411,43 +1422,68 @@ func getPhase(pod *v1.Pod, info []v1.ContainerStatus, podIsTerminal bool) v1.Pod
 		}
 	}
 
+	// counters for sidecar and main containers
 	unknown := 0
 	running := 0
 	waiting := 0
 	stopped := 0
 	succeeded := 0
-	for _, container := range spec.Containers {
-		containerStatus, ok := podutil.GetContainerStatus(info, container.Name)
-		if !ok {
-			unknown++
-			continue
-		}
+	terminatedSidecar := 0
+	crashedSidecar := 0
+	hasInitialized := false
 
-		switch {
-		case containerStatus.State.Running != nil:
-			running++
-		case containerStatus.State.Terminated != nil:
-			stopped++
-			if containerStatus.State.Terminated.ExitCode == 0 {
-				succeeded++
+	checkContainers := func(containers []v1.Container, isInitContainers bool) {
+		for _, container := range containers {
+			if isInitContainers && !kubetypes.IsSidecarContainer(&container) {
+				continue
 			}
-		case containerStatus.State.Waiting != nil:
-			if containerStatus.LastTerminationState.Terminated != nil {
+
+			containerStatus, ok := podutil.GetContainerStatus(info, container.Name)
+			if !ok {
+				unknown++
+				continue
+			}
+
+			if !isInitContainers {
+				if containerStatus.State.Running != nil || containerStatus.State.Terminated != nil {
+					hasInitialized = true
```

Comment:
Adding a comment here would be useful. Not clear to me from the name why this is updated to true on the basis of a single container's state.